src/handlers/check_commits/no_mentions.rs

```diff
@@ -1,37 +1,29 @@
 //! Purpose: When opening a PR, or pushing new changes, check for github mentions
 //! in commits and notify the user of our no-mentions in commits policy.
 
-use std::fmt::Write;
-
 use crate::{config::NoMentionsConfig, github::GithubCommit};
 
 pub(super) fn mentions_in_commits(
     _conf: &NoMentionsConfig,
     commits: &[GithubCommit],
 ) -> Option<String> {
-    let mut mentions_commits = Vec::new();
-
-    for commit in commits {
-        if !parser::get_mentions(&commit.commit.message).is_empty() {
-            mentions_commits.push(&*commit.sha);
-        }
-    }
+    let mentions_commits = commits
+        .into_iter()
+        .filter(|c| !parser::get_mentions(&c.commit.message).is_empty())
+        .map(|c| format!("    - {}\n", c.sha))
+        .collect::<String>();
 
     if mentions_commits.is_empty() {
         None
     } else {
-        Some(mentions_in_commits_warn(mentions_commits))
-    }
-}
-
-fn mentions_in_commits_warn(commits: Vec<&str>) -> String {
-    let mut warning = format!("There are username mentions (such as `@user`) in the commit messages of the following commits.\n  *Please remove the mentions to avoid spamming these users.*\n");
-
-    for commit in commits {
-        let _ = writeln!(warning, "    - {commit}");
+        Some(
+            String::from(
+                r"There are username mentions (such as `@user`) in the commit messages of the following commits.
+  *Please remove the mentions to avoid spamming these users.*
+",
+            ) + &mentions_commits,
+        )
     }
-
-    warning
 }
```

Comment:
Why all of these unrelated changes? Can you avoid them and revert back to the way it was before.

Reply:
I noticed that there were some unnecessary allocations. This was just a refactor to make it more optimal :) happy to revert.

Reply:
Hum, I see, I though it was purely a stylistic choice, happy to discuss those in a follow up pr.

However I liked the added test in this file, can you bring it back?